930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/rancher/RancherHelper.java

```
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.delegate.task.k8s.rancher;
+
+import static io.harness.data.structure.EmptyPredicate.isNotEmpty;
+
+import io.harness.annotations.dev.CodePulse;
+import io.harness.annotations.dev.HarnessModuleComponent;
+import io.harness.annotations.dev.ProductModule;
+import io.harness.delegate.beans.connector.rancher.RancherConnectorBearerTokenAuthenticationDTO;
+import io.harness.delegate.task.k8s.RancherK8sInfraDelegateConfig;
+import io.harness.k8s.model.KubernetesConfig;
+
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+@UtilityClass
+public class RancherHelper {
+  public String getRancherUrl(RancherK8sInfraDelegateConfig rancherK8sInfraDelegateConfig) {
+    if (rancherUrlExists(rancherK8sInfraDelegateConfig)) {
+      return rancherK8sInfraDelegateConfig.getRancherConnectorDTO().getConfig().getConfig().getRancherUrl();
+    }
+    return StringUtils.EMPTY;
+  }
+
+  public String getRancherBearerToken(RancherK8sInfraDelegateConfig rancherK8sInfraDelegateConfig) {
+    if (rancherBearerTokenExists(rancherK8sInfraDelegateConfig)) {
+      return String.valueOf(
+          ((RancherConnectorBearerTokenAuthenticationDTO) rancherK8sInfraDelegateConfig.getRancherConnectorDTO()
+                  .getConfig()
+                  .getConfig()
+                  .getCredentials()
+                  .getAuth())
+              .getPasswordRef()
+              .getDecryptedValue());
+    }
+    return StringUtils.EMPTY;
```

Is there a scenario when the URL can be empty? If not I think that we shouldn't return EMPTY (as it will fail later), we should throw an exception with the right message. The same is for the KubeConfigTokenName

---

Yes, I thought of this while implementing as well. There can't be a scenario as of now where these values are empty, since we enforce them on connector level itself. Even if it does end up coming as empty somehow, there are 2 usages of this:
- Token cleanup
- Token generation

Token generation handles exceptions and logs them properly. 
For token cleanup, all exceptions are ignored and are just logged (on debug level). Our K8s deletions like resource deletions are all ignoring deletion errors, so wanted to keep it like that. We don't really want to fail the deployment in case we are not able to cleanup the token from rancher. 
For KubeConfigTokenName, that is used in cleanup only. If it is empty somehow, cleanup is skipped.